test/models/submission/test_run_test.rb

```diff
@@ -101,5 +101,20 @@ class Submission::TestRunTest < ActiveSupport::TestCase
     assert_equal test_as_hash, result.as_json(1, 2, 3) # Test with arbitary args
   end
 
+  test "tooling_job" do
+    submission = create :submission
+    job = create_test_runner_job!(submission)
+    Submission::TestRun::Process.(job)
+    test_run = submission.test_run
+
+    Exercism::ToolingJob.expects(:new).with(test_run.tooling_job_id, {}).returns(job)
+    job.expects(:stdout)
+    job.expects(:stderr)
+    job.expects(:metadata)
+    test_run.stdout
+    test_run.stderr
+    test_run.metadata
```

Comment:
Is this just testing whether this doesn't throw?

Reply:
It's testing:
1. The methods are delegated correctly
2. The methods exist (because stubbing missing methods in our testing setup raises an exception)